### PR TITLE
feat: encryption keys in database

### PR DESCRIPTION
**Pull request by livio-a**

### CODE REVIEW

File: cmd/admin/start/encryption_keys.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+	encryptionKeys.OIDCKey = []byte(key)
+	encryptionKeys.OTP, err = crypto.NewAESCrypto(keyConfig.OTP, keyStorage)
+	if err != nil {
+		return nil, err
+	}
+	encryptionKeys.SMS, err = crypto.NewAESCrypto(keyConfig.SMS, keyStorage)
+	if err != nil {
+		return nil, err
+	}
+	encryptionKeys.SMTP, err = crypto.NewAESCrypto(keyConfig.SMTP, keyStorage)
```
Comment (fgerschwiler) [inline cmd/admin/start/encryption_keys.go:71]: same as with the sms key?

File: internal/crypto/file/file.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+		return nil, fmt.Errorf("key no found")
+	}
+	return &crypto.Key{
+		ID:    id,
+		Value: key,
+	}, nil
+}
+
+func (d *Storage) CreateKeys(keys ...*crypto.Key) error {
+	return fmt.Errorf("unimplemented") //TODO: ?
```
Comment (fgerschwiler) [inline internal/crypto/file/file.go:43]: todo? if I understand it correct this method should not be used, so maybe remove todo and throw error like "create keys should not be called in this provider"

File: cmd/admin/start/encryption_keys.go [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+	key, err := crypto.LoadKey(keyConfig.OIDC.EncryptionKeyID, keyStorage)
+	if err != nil {
+		return nil, err
+	}
+	encryptionKeys.OIDCKey = []byte(key)
+	encryptionKeys.OTP, err = crypto.NewAESCrypto(keyConfig.OTP, keyStorage)
+	if err != nil {
+		return nil, err
+	}
+	encryptionKeys.SMS, err = crypto.NewAESCrypto(keyConfig.SMS, keyStorage)
```
[the remark this answers is not in the document]
Comment (livio-a) [reply cmd/admin/start/encryption_keys.go:67]: we don't know if a provider is set and to create a new one will need the encryptionKey as well

File: cmd/admin/start/encryption_keys.go (hunk 1 above)
Comment (livio-a) [reply cmd/admin/start/encryption_keys.go:71]: see above

File: internal/crypto/file/file.go (hunk 2 above)
Comment (livio-a) [reply internal/crypto/file/file.go]: fixed

**Review by fgerschwiler — APPROVED** [review]

**Pull request merged**